### DISCUSSION TITLE
REST API: List collections with "fields control"

### DISCUSSION THREAD

**Discussion by jekuer** [directus/directus#2745] [Archive]
I want to get a list of all my collections (via REST API) in the most shortest way to speed things up - basically only the names.

The Items endpoint provides me with an option to limit the fields I get with the response.
It would be cool to have a similar option with the collections endpoint.

Not sure if this will be addressed with v9, but could not find information about it.

**Comment by benhaynes** [comment]
I'm not sure I understand the use-case. The `fields` data on the [Collections](https://docs.directus.io/api/collections.html) endpoint is meant to give all of the info required by the App... and I don't know how much of a speed improvement there would be for limiting these (unless there were a LOT of fields).

Is this a performance or permissions issue? Could you explain a bit more about the reasoning?

**Reply by jekuer** [reply]
This is about performance.
And about GET only.

I have an extremely small setup.
Going for GET /:project/collections gives me a long json (87kb).
![is](https://user-images.githubusercontent.com/8572883/83881137-7cb1e400-a740-11ea-8dc9-6fae047e44e8.png)

If I could skip the fields, just getting the collection name (and maybe note, hidden, ...), it would lead to a list at 1,3kb.
![toBe](https://user-images.githubusercontent.com/8572883/83881132-7c194d80-a740-11ea-8ebb-1792cba46808.png)

**Use Case:**
I might have a collection for pages and a collection for blog posts. If I want to build a frontend, which should work with or without the blog section, it could make sense to check if there is a collection for it.
I know there are plenty of other ways to accomplish this, but I was wondering, why I can minimize the response on most calls, but not /collections.

**Reply by benhaynes** [reply]
Makes sense, and seems like something easy enough to include in v9.

@directus/team ?

**Reply by rijkvanzanten** [reply]
Yeah! We're planning on supporting the same query parameters across all endpoints
